lib/Sema/TypeCheckAccess.cpp

```diff
@@ -1471,6 +1453,30 @@ class UsableFromInlineChecker : public AccessControlCheckerBase,
 class ExportabilityChecker : public DeclVisitor<ExportabilityChecker> {
   class Diagnoser;
 
+  // Problematic origin of an exported type.
+  //
+  // This enum must be kept in sync with
+  // diag::decl_from_hidden_module and
+  // diag::conformance_from_implementation_only_module.
+  enum class ExportedOriginKind : uint8_t {
```

Comment:
nitpick: a more descriptive name can be `DisallowedExportedOriginKind`

Reply:
Good point, to keep it short I went with `DisallowedOriginKind` as the exportability concept is already in the parent type name.